servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -283,25 +285,22 @@ public void cleanupUser(long userId) throws IOException {
     // is considered "dead" may still be using or committing this block.
     // A user may have multiple temporary directories for temp blocks, in diffrent StorageTier
     // and StorageDir.
-    Set<String> dirs = new HashSet<String>(tempBlocksToRemove.size());
     for (TempBlockMeta tempBlockMeta : tempBlocksToRemove) {
       String fileName = tempBlockMeta.getPath();
-      try {
-        String dirName = CommonUtils.getParent(fileName);
-        dirs.add(dirName);
-      } catch (InvalidPathException e) {
-        LOG.error("Error in cleanup userId {}: cannot parse parent dir of {}", userId, fileName);
-      }
       if (!new File(fileName).delete()) {
         LOG.error("Error in cleanup userId {}: cannot delete file {}", userId, fileName);
       } else {
         removedTempBlocks.add(tempBlockMeta.getBlockId());
       }
     }
-    // TODO: Cleanup the user folder across tiered storage.
-    for (String dirName : dirs) {
-      if (!new File(dirName).delete()) {
-        LOG.error("Error in cleanup userId {}: cannot delete directory {}", userId, dirName);
+
+    // Go through all the storage directories and delete the user folders which should be empty
+    for (StorageTier tier : mMetaManager.getTiers()) {
+      for (StorageDir dir : tier.getStorageDirs()) {
+        File userFolder = new File(CommonUtils.concatPath(dir.getDirPath(), userId));
+        if (userFolder.exists() && !userFolder.delete()) {
+          LOG.error("Failed to clean up user: {} with directory: {}", userId, userFolder.getPath());
```

Comment:
Please javadoc that, deleting this user dir is better, but leave it there will not affect the correctness---just less clean. That's why we LOG.error instead of throwing exceptions.